pkg/zfs/volume.go

```diff
@@ -256,3 +260,44 @@ func RemoveSnapFinalizer(snap *apis.ZFSSnapshot) error {
 	_, err := snapbuilder.NewKubeclient().WithNamespace(OpenEBSNamespace).Update(snap)
 	return err
 }
+
+// RemoveBkpFinalizer removes finalizer from ZFSBackup CR
+func RemoveBkpFinalizer(bkp *apis.ZFSBackup) error {
+	bkp.Finalizers = nil
+
+	_, err := bkpbuilder.NewKubeclient().WithNamespace(OpenEBSNamespace).Update(bkp)
+	return err
+}
+
+// UpdateBkpInfo updates the backup info with the status
+func UpdateBkpInfo(bkp *apis.ZFSBackup, status apis.ZFSBackupStatus) error {
```

Comment:
q_: Why are the finalizers being added to backup and not restore resources?

Reply:
as part of backup request, we have a dependant object, which is the snapshot. So when you clean up that, snapshot should also gets cleaned up. There is no dependant object for restore.